generate_makefile.bash

```diff
@@ -204,7 +209,10 @@ KOKKOS_SRC_PATH=${KOKKOS_PATH}
 KOKKOS_SETTINGS="KOKKOS_SRC_PATH=${KOKKOS_SRC_PATH}"
 #KOKKOS_SETTINGS="KOKKOS_PATH=${KOKKOS_PATH}"
 
-if [ ${#COMPILER} -gt 0 ]; then
+if [ ${#COMPILER} -gt 0 ] && [ ${#KOKKOS_DEVICES} -gt 0 ]; then
```

Comment:
@dalabre from our PR review, change this `if` to an `elif` and check that the `COMPILER` string is empty in that elif check, not greater than 0;
re-include the original check as the "if" case before this, i.e.
```
if [ ${#COMPILER} -gt 0 ]; then
  KOKKOS_SETTINGS="${KOKKOS_SETTINGS} CXX=${COMPILER}"
```

You may also need to set `COMPILER=""` at the start of the script, but check without doing that first. 

It's getting close :)

Reply:
@ndellingwood  Corrected implementation; testing thus far shows correct behavior.
Testing continues ( have a couple dozen cases to look at ); will post updated pull request tomorrow.